mlog/logr.go

```diff
@@ -39,30 +39,32 @@ type LogTargetCfg map[string]*LogTarget
 type LogrCleanup func() error
 
 func newLogr(targets LogTargetCfg) (*logr.Logger, error) {
-	var errs error
-
-	lgr := logr.Logr{}
-
+	lgr := &logr.Logr{}
 	lgr.OnExit = func(int) {}
 	lgr.OnPanic = func(interface{}) {}
-
 	lgr.OnLoggerError = onLoggerError
 	lgr.OnQueueFull = onQueueFull
 	lgr.OnTargetQueueFull = onTargetQueueFull
 
+	err := logrAddTargets(lgr, targets)
+	logger := lgr.NewLogger()
+	return &logger, err
```

Comment:
Are we returning a logger also in case of error cause we accept the fact that adding some targets might fail?

Reply:
Right.  An individual target failing is not fatal.  We want to log that failure on the targets that were successful.  For example, if we cannot connect to the syslog server we want to log that error to file.

Reply:
BTW, the error returned may be a multi-error, one for each target that failed to load.